packages/mdc-select/index.js

```diff
@@ -214,29 +286,148 @@ class MDCSelect extends MDCComponent {
+      },
+      setValue: (value) => {
+        const element = /** @type {HTMLElement} */ (this.menuElement_.querySelector(`[value="${value}"]`));
+        if (element) {
+          this.setEnhancedSelectedIndex_(this.menu_.items.indexOf(element));
+        } else {
+          this.selectedText_.textContent = value;
+        }
+      },
+      openMenu: () => {
+        if (this.menu_ && !this.menu_.open) {
+          this.menu_.open = true;
+          this.menuOpened_ = true;
+        }
+      },
+      closeMenu: () => {
+        if (this.menu_ && this.menu_.open) {
+          this.menu_.open = false;
+        }
+      },
+      isMenuOpen: () => this.menu_ && this.menuOpened_,
+      setSelectedIndex: (index) => {
+        this.setEnhancedSelectedIndex_(index);
+      },
+      setDisabled: (isDisabled) => {
+        this.selectedText_.setAttribute('tabindex', isDisabled ? '-1' : '0');
+        isDisabled ? this.root_.classList.add(cssClasses.DISABLED) : this.root_.classList.remove(cssClasses.DISABLED);
+      },
+    };
+  }
+
+  /**
+   * Sets the selected index of the enhanced menu.
+   * @param {number} index
+   * @private
+   */
+  setEnhancedSelectedIndex_(index) {
+    const selectedItem = this.menu_.items[index];
+    this.selectedText_.textContent = selectedItem ? selectedItem.textContent.trim() : '';
+    const previouslySelected = this.menuElement_.querySelector(strings.SELECTED_ITEM_SELECTOR);
+
+    if (previouslySelected) {
+      previouslySelected.classList.remove(cssClasses.SELECTED_ITEM_CLASS);
+      previouslySelected.removeAttribute('aria-selected');
+    }
+
+    if (selectedItem) {
+      selectedItem.classList.add(cssClasses.SELECTED_ITEM_CLASS);
+      selectedItem.setAttribute('aria-selected', 'true');
+    }
+
+    this.layout();
+  }
+
+  /**
+   * @return {!{
+   *   addClass: function(string): void,
+   *   removeClass: function(string): void,
+   *   hasClass: function(string): void,
+   *   isRtl: function(): boolean,
+   *   setRippleCenter: function(number): void,
+   *   activateBottomLine: function(): void,
+   *   deactivateBottomLine: function(): void,
+   *   notifyChange: function(!{value: string}): void
+   * }}
+   * @private
+   */
+  getCommonAdapterMethods_() {
+    return {
+      addClass: (className) => this.root_.classList.add(className),
+      removeClass: (className) => this.root_.classList.remove(className),
+      hasClass: (className) => this.root_.classList.contains(className),
+      isRtl: () => window.getComputedStyle(this.root_).getPropertyValue('direction') === 'rtl',
+      setRippleCenter: (normalizedX) => this.lineRipple_ && this.lineRipple_.setRippleCenter(normalizedX),
+      activateBottomLine: () => this.lineRipple_ && this.lineRipple_.activate(),
+      deactivateBottomLine: () => this.lineRipple_ && this.lineRipple_.deactivate(),
+      notifyChange: (evtData) => {
+        evtData.index = this.selectedIndex;
+        this.emit(strings.CHANGE_EVENT, evtData);
```

Comment:
I think we should make this bubble to match native `change` which we're effectively replacing.

Reply:
If we change this to emit the `change` event, the native select will emit 2 change events (because this is also emitted during the handleChange event and the native one) . By using our custom change event, it allows you to use the same event for both implementations while only getting 1 event.

Reply:
I think I clarified this offline, but I was referring to the bubbling behavior, not the event name. (Currently this event doesn't bubble, but native `change` does.)